Virology Rework

XDTM opened the pull request:
:cl: XDTM
add: Added four new symptoms: Regeneration and Tissue Regrowth heal respectively brute and burn damage slowly; Flesh Mending and Heat Resistance are the level 8 version of those symptoms, and heal faster.
del: The Toxic Compensation, Toxic Metabolism and Stimulants symptoms have been removed.
del: Viruses no longer stack. Viruses can be overridden if the infecting virus has higher transmittability than the previous virus' resistance.
tweak: Viruses can now have 8 symptoms, up from 6.
/:cl:

Based off of suggestions in coderbus.
Changes:
-Advanced viruses **no longer stack. Limit of one.**
-Viruses can now have up to 8 symptoms. Stealth threshhold of the Pandemic has been raised to 4 to compensate.
-Viruses can now take over other advance viruses on spread, if their Transmittability is higher than the defending virus' Resistance.
-Some viruses no longer use chems, but the effects are mostly unchanged.
-Removed Toxic Compensation, Toxic Metabolism, Anti-Bodies Metabolism and Stimulants.
-Added Regeneration and Flesh Mending, brute-healing symptoms. Flesh Mending is level 8 and also turns cloneloss into burn damage.
-Added Tissue Regrowth and Heat Resistance, burn-healing symptoms. Heat Resistance is level 8 and also helps stabilize body temperature.
-SYMTOM_ACTIVATION_PROB has been increased to 5% from 3% to make up for the loss of stacking.
-Healing procs every tick for 1 damage (2 for the upgraded ones).

XDTM committed:
Maybe this time it'll work?

XDTM committed:
What about now

XDTM committed:
mayyyybe now?

ChemicalRascal commented:
This is waaaaay more sensible. Adding difficulty and complexity, rather than outright removal.

Healing viruses are an important part of non-antags dealing with the late-game, but having stimulant not be part of that lategame will probably be beneficial in its own regard.

CyberbossJHCB commented on code/datums/diseases/advance/symptoms/stimulant.dm:
```diff
@@ -1,38 +0,0 @@
-	resistance = -3
-	stage_speed = -2
-	transmittable = -4
-	level = 3
```
If people think stim is so OP why not at least try making it a level 8 symptom before removing it entirely

ChemicalRascal replied:
Maybe if it was simply split into two symptoms, as another alternative? Put the sanic-speed-gotta-go-fast in one, and the stun reduction in another. That'd certainly lead to virologists having to make important choices, simply due to the increased number of symptoms.

Non-antag viros have no need to give stun reduction to anyone, while heads and sec should have boosted footspeed. Antag viros, on the other hand, simply won't be able to make their ideal viruses.

CyberbossJHCB replied:
I agree with you. But it would require a rewrite of how it functions fundamentally, given that all the symptom does is give you shots of ephedrine.

ChemicalRascal replied:
Oh, right.

Ergovisavi replied:
> If people think stim is so OP why not at least try making it a level 8 symptom before removing it entirely

Because it's the way it affects the game. By potentially spreading a massive movespeed increase and stun resistance amongst the crew we get into situations where fundamental parts of gameplay for SS13 are affected, IE: The stun based combat. The other aspect is the fact that it is indefinite in duration. A pill of ephedrine is not particularly damning on the system, as there is an extreme potential for it to be entirely useless when used in context due to timing, etc. A stimulant virus is on 24/7 and drags down the game as a whole.

Not that SS13's ridiculous stun base combat is good, but having something in the game that permanently fucks with the game state in that fashion (on top of just being a fucking boring mechanic) isn't kosher.

Cyberboss reviewed (commented):
I suppose I had this coming for making airborne Stim, SelfResp, ToxFilt, ToxComp 5 minutes into every round when I play viro, but still...

Aloraydrel commented:
This upsets the powergamer

jaxy15 commented:
kevinz000 reacted with thumbs down emoji

XDTM committed:
Stat nerfs

chickenboy10 commented:
This is much better PR then the straight up removal PR, stimulant was a bit stupidly overpowered for a level 3 symptom.
Alsothe splitting of brute/burn/toxic damage heals makes the viros job harder, both in managing resources and picking the symptoms they need.

Shadowlight213 commented on code/datums/diseases/advance/symptoms/heal.dm:
```diff
@@ -64,22 +64,187 @@ Bonus
+
+	if(M.getCloneLoss() > 0)
+		M.adjustCloneLoss(-1)
+		M.take_bodypart_damage(0, 2)
```
So, the symptoms now also heal cloneloss??
From looking at the code, with the way this works, this part is pretty much pointless.
The minimum possible amount healed is 2, and this converts it into 2 damage.
Right after this, the symptom does the healing, mean at the minimum it would just heal cloneloss that tick, with no increase in burn/brute and at the max heal only 2 damage plus healing the 1 cloneloss.
It would be far more efficient to simply have it heal the cloneloss, then subtract 2 from the heal_amt var.

XDTM replied:
The damage dealt is Burn, not Brute, so to completely heal it you need both symptoms.

Shadowlight213 replied:
I see.
The fact that it's just a number denoting the damage type is silly. Not your fault though

XDTM replied:
Added a comment to avoid future confusion.

Shadowlight213 reviewed (changes requested):
Cloneloss healing code is inefficient. The damaging just to heal right after is unneeded.

Shadowlight213 commented:
Actually.. looking at this, this is a buff to the heal virus disguised as a nerf, considering toxin damage is relatively rare.
2 symptom slots with advanced, and you now heal twice as fast as the version before, with the bonus of healing cloneloss.
I applaud your sneakiness good sir.

kevinz000 commented:
Shadow light pls

XDTM commented:
In case you didnt notice the proc chance has been halved, which in turn halves the healing received on average. And the values are those of toxic compensation. 
In any case i have a full-fledged rework incoming, so dont look into these changes too far yet.

XDTM committed:
VIRUS REWORK

XDTM committed:
Missed the files

XDTM committed:
Less heal

XDTM committed:
Nope, stims don't stay

XDTM committed:
Admin virus

phil235 commented on code/datums/diseases/_MobProcs.dm:
```diff
@@ -32,23 +29,30 @@
+		if(istype(D, /datum/disease/advance))
+			var/datum/disease/advance/DD = D
+			if (P.totalResistance() < DD.totalTransmittable()) //Overwrite virus if the attacker's Transmission is lower than the defender's Resistance. This does not grant immunity to the lost virus.
+				P.remove_virus()
```
if we can only ever have one virus at a time, then the viruses var no longer needs to be a list.

XDTM committed:
More small changes

ChemicalRascal commented on code/datums/diseases/advance/symptoms/genetics.dm:
```diff
@@ -29,7 +29,7 @@ Bonus
 /datum/symptom/genetic_mutation/Activate(datum/disease/advance/A)
 	..()
-	if(prob(SYMPTOM_ACTIVATION_PROB * 5)) // 15% chance
+	if(prob(SYMPTOM_ACTIVATION_PROB * 3)) // 15% chance
```
Is this still Pr(Activate) = 0.15, as the comment implies?

XDTM replied:
Yeah, i increased SYMPTOM_ACTIVATION_PROB to 5% to make up for the loss of stacking.

kevinz000 commented on code/modules/reagents/chemistry/reagents/virus_reagents.dm:
```diff
@@ -0,0 +1,198 @@
+/datum/reagent/virus/fleshdeath //heparin + lipolicide
+	name = "Viral Necrotic Agent"
+	id = "viral_fleshdeath"
+	description = "Increases stun resistance."
```
um

XDTM replied:
Oops

XDTM committed:
Minor text fixes

optimumtact commented:
viruses creating reagents to do healing is just stupid, just move their effects into the virus?

XDTM commented:
Will do

XDTM committed:
New Patch

XDTM commented:
-Now most viruses don't use reagents and those that do use normal ones, like ethanol for sensory destruction. Removed virus chems.
-Fixed overlay spam on the heal symptoms.
-Heal symptoms now heal by heal_amt in total and not per damaged limb.
-Increased Pandemic Stealth detection due to symptom limit increase.

XDTM commented:
It does, yes

ChemicalRascal commented:
Surely that's not intentional, then? Stealth: 2 has HIDDEN_SCANNER, Stealth: 4 has HIDDEN_SCANNER, Stealth: 3 does not?

XDTM commented:
oh i see the issue, i'll fix that now, sorry

XDTM committed:
3

optimumtact commented:
Are we ready to go here?

XDTM commented:
The code is ready

AnturK commented on tgstation.dme:
```diff
@@ -14,7 +14,7 @@
 // BEGIN_INCLUDE
 #include "_maps\__MAP_DEFINES.dm"
-#include "_maps\tgstation2.dm"
+#include "_maps\ministation.dm"
```
Unrelated.

XDTM committed:
Mini best map
